src/DataProtection/Cryptography.Internal/src/CryptoUtil.cs

```diff
@@ -76,12 +76,23 @@ public static Exception Fail(string message)
 #endif
         public static bool TimeConstantBuffersAreEqual(byte* bufA, byte* bufB, uint count)
         {
+#if NETCOREAPP
+            ReadOnlySpan<byte> bytesA, bytesB;
+            var byteCount = Convert.ToInt32(count);
+            unsafe
+            {
+                bytesA = new ReadOnlySpan<byte>(bufA, byteCount);
+                bytesB = new ReadOnlySpan<byte>(bufB, byteCount);
+            }
+            return CryptographicOperations.FixedTimeEquals(bytesA, bytesB);
```

Comment:
❤️ 

There are a bunch of places in data protection where we could leverage new primitives introduced to the framework in the past few releases. Might be interesting to have an issue for that. Identity likely can also benefit from it.